MetaMask/metamask-extension
# Adding Modal component

## Explanation
Adds new `Modal` component that uses context to provide state to the sub components from the parent component. This also updates the `ModalContent` component so options are only set once and then handled through the `Modal` context

* Fixes #15432

Dependencies
- https://github.com/MetaMask/metamask-extension/pull/19034
- https://github.com/MetaMask/metamask-extension/pull/18979

## Screenshots/Screencaps

### Before
Current `ModalContent`

https://github.com/MetaMask/metamask-extension/assets/8112138/9bcd9354-b67e-403a-9e32-1c78a78365d5


### After
Updated `ModalContent`

https://github.com/MetaMask/metamask-extension/assets/8112138/f7ae7423-628d-4a2e-9dc3-6534437b3f86


New `Modal` component and documentation


https://github.com/MetaMask/metamask-extension/assets/8112138/9615b660-81f9-4292-a29c-60638dea42e9


## Manual Testing Steps

<!--
How should reviewers and QA manually test your changes? For instance:

- Go to the storybook build in this PR
- Search `Modal` component and go to one in the Component Library
- Check controls, stories and documentation

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
